File: src/test/console/java/src/test/java/kobiton/com/testng/AndroidAppTest.java
package kobiton.com.testng;

import com.kobiton.service.DeviceService;
import com.kobiton.service.model.Device;
import java.util.concurrent.TimeUnit;
import org.openqa.selenium.By;
import org.openqa.selenium.Dimension;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterTest;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;
import io.appium.java_client.android.AndroidDriver;
import java.io.IOException;
import kobiton.com.utils.AutomationUtils;

public class AndroidAppTest {

    public static AndroidDriver<WebElement> driver = null;
    public final String question = "Acura MDX";

    @BeforeTest
    public void Setup() {
        try {
            Device device = new DeviceService().getOnlineDevice("Android");
            System.out.println(String.format("Execute with: %s  udid: %s", device.getDeviceName(), device.getUdid()));
            driver = new AndroidDriver<>(AutomationUtils.kobitonServerUrl(), AutomationUtils.desiredCapabilitiesAndroidApp(device));
            driver.manage().timeouts().implicitlyWait(60, TimeUnit.SECONDS);
            String kobitonSessionId = driver.getSessionDetails().get("kobitonSessionId").toString();
            System.out.println("https://portal.kobiton.com/sessions/" + kobitonSessionId);
        } catch (IOException ex) {
            Assert.fail("set up failed", ex);
        }
    }

    @AfterTest
    public void Teardown() {
        try {
            if (driver != null) {
                driver.quit();
            }
        } catch (Exception e) {
            //Do-nothing
        }
    }

    @Test(priority = 1, description = "should verify Acura categories")
    public void testSearchQuestionsOnAcuraSupportCommunity() {

        /*
      * Steps:
      * 1. Click on "Car and Truck" Categories on Homepage
      * 2. Click on "Acura" Categories
      *
      * Expected:
      * General Information is "Acura".
        */

        driver.findElementByXPath("//*[@resource-id='android:id/home']").click();
        sleep(3);
        driver.findElementByXPath("//*[@text='Car and Truck']").click();
        driver.findElementByXPath("//*[@text='Acura']").click();
        sleep(3);
        String acuraText = driver.findElementByXPath("//*[@resource-id='com.dozuki.ifixit:id/topic_title' and @index=1]")
                .getText();

        Assert.assertEquals(acuraText, "Acura");

    }

    public void sleep(int seconds) {
        try {
            Thread.sleep(seconds * 1000);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }
}

File: src/test/console/dotnet/MobileAutomation/Tests/Scripts/AndroidAppTest.cs
using System;
using System.Threading;
using MobileAutomation.Config;
using MobileAutomation.Service;
using MobileAutomation.Service.Model;
using MobileAutomation.Tests.Utils;
using OpenQA.Selenium.Remote;
using Xunit;

namespace MobileAutomation.Tests.Scripts
{
  public class AndroidAppTest : BaseTest
  {
    public AndroidAppTest()
    {
      var device = new DeviceService().GetOnlineDeviceSync(DeviceType.Android);
      Console.WriteLine(string.Format("Device {0} osVersion: {1} udid: {2}",
                                      device.DeviceName,
                                      device.PlatformVersion,
                                      device.Udid));
      driver = new RemoteWebDriver(AutomationUtils.Instance.KobitonServerUrl,
                                   AutomationUtils.Instance.DesiredCapabilitiesAndroidApp(device),
                                   TimeSpan.FromSeconds(Configs.SESSION_TIMEOUT));
      driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromSeconds(60);
    }

    [Fact]
    public void TestIFixitAndroidApp()
    {
      try {
        TestSearchQuestionsOnAcuraSupportCommunity();
      }finally {
        this.Dispose();
      }
    }

    public void TestSearchQuestionsOnAcuraSupportCommunity()
    {
      Console.WriteLine("should allow to search some questions on Acura Support Community");
      /*
       * Steps:
       * 1. Click on "Car and Truck" Categories on Homepage
       * 2. Click on "Acura" Categories
       *
       * Expected:
       * General Information is "Acura"
       */
      Thread.Sleep(2000);
      driver.FindElementByXPath("//*[@resource-id='android:id/home']").Click();
      Thread.Sleep(2000);
      driver.FindElementByXPath("//*[@text='Car and Truck']").Click();
      driver.FindElementByXPath("//*[@text='Acura']").Click();
      Thread.Sleep(2000);
      string acuraText = driver
        .FindElementByXPath("//*[@resource-id='com.dozuki.ifixit:id/topic_title' and @index=1]").Text;
      Assert.Equal(acuraText, "Acura");
    }
  }
}